### DISCUSSION TITLE
How do I redirect a command output to file in custom rules?

### DISCUSSION THREAD

**Discussion by xq114** [xmake-io/xmake#1775] [Q&A] [answered]
batchcmds貌似不支持iorunv，如何把命令的执行输入重定向到文件？

**Comment by waruqi** [comment, marked as answer]
batchcmds 是序列化的 cmd list，并没有立即执行，无法去 iorunv的返回值，可以走 `runv(program, argv, {stdout = "", stderr = "", stdin = ""})` 。execv 也可以

stdout 可以传递 文件路径，或者 io.open的文件对象，或者 pipe 对象。。都可以实现重定向

**Reply by xq114** [reply]
实测runv不行（或许因为runv本来就没有输出），execv可以

**Reply by waruqi** [reply]
runv内部重定向已经捕获了
